exporter/containerimage/exptypes/types.go

```diff
@@ -23,3 +40,64 @@ type Platform struct {
 	ID       string
 	Platform ocispecs.Platform
 }
+
+func ExporterAnnotationIndexKey(key string) string {
+	return exporterAnnotationKey(AnnotationIndex, nil, key)
+}
+
+func ExporterAnnotationIndexDescriptorKey(key string) string {
+	return exporterAnnotationKey(AnnotationIndexDescriptor, nil, key)
+}
+
+func ExporterAnnotationManifestKey(p *ocispecs.Platform, key string) string {
+	return exporterAnnotationKey(AnnotationManifest, p, key)
+}
+
+func ExporterAnnotationManifestDescriptorKey(p *ocispecs.Platform, key string) string {
+	return exporterAnnotationKey(AnnotationManifestDescriptor, p, key)
+}
+
+func exporterAnnotationKey(annotationType string, platform *ocispecs.Platform, key string) string {
+	switch annotationType {
+	case AnnotationManifest, AnnotationManifestDescriptor:
+	case AnnotationIndex, AnnotationIndexDescriptor:
+		platform = nil
+	default:
+		panic("unknown annotation type")
+	}
+
+	if platform == nil {
+		return fmt.Sprintf("annotation-%s.%s", annotationType, key)
+	}
+	return fmt.Sprintf("annotation-%s[%s].%s", annotationType, platforms.Format(*platform), key)
+}
+
+func ExporterAnnotationKeyInvert(result string) (string, *ocispecs.Platform, string, error) {
```

Comment:
Invert? `ParseAnnotationKey()` ?  It could also return typed `(AnnotationKey, bool, error)` and you don't need `ExporterIsAnnotationKey` as well. `AnnotationKey` can implement stringer if it helps.